TreeView: add the "getting started" article (#2591)
DevExpress/devextreme-documentation: concepts/05 UI Components/TreeView/00 Getting Started with TreeView/05 Bind the TreeView to Data.md

> To bind the TreeView to a local array, use the [dataSource](/Documentation/ApiReference/UI_Components/dxTreeView/Configuration/#dataSource) property. The following articles explain how to do this for other storage types:

Reviewer: "to do this" == "to bind the TreeView to data in other storage types"?

> To bind the TreeView to a local array, use the [dataSource](/Documentation/ApiReference/UI_Components/dxTreeView/Configuration/#dataSource) property. The following articles explain how to do this for other storage types:

Reviewer 2: ```suggestion
To bind the TreeView to a local array, use the [dataSource](/Documentation/ApiReference/UI_Components/dxTreeView/Configuration/#dataSource) property. Refer to the articles listed below for instructions on how to do this for other storage types:
```

            categoryId: "1_1_2",
            name: "SuperLCD 70",
            image: IMAGE_URL + "9.png",
            price: 4000
        }, {
            ID: "1_1_3",
            categoryId: "1_1",
            name: "Monitors"
        }, {
            ID: "1_1_3_1",
            categoryId: "1_1_3",
            name: "19\"",
        }, {
            ID: "1_1_3_1_1",
            categoryId: "1_1_3_1",
            name: "DesktopLCD 19",
            image: IMAGE_URL + "10.png",
            price: 160
        }, {
            ID: "1_1_4",
            categoryId: "1_1",
            name: "Projectors"
        }, {
            ID: "1_1_4_1",
            categoryId: "1_1_4",
            name: "Projector Plus",
            image: IMAGE_URL + "14.png",
            price: 550
        }, {
            ID: "1_1_4_2",
            categoryId: "1_1_4",
            name: "Projector PlusHD",
            image: IMAGE_URL + "15.png",
            price: 750
        }
    ];

    @Injectable({
        providedIn: 'root'
    })
    export class ProductsService {
        getProducts(): Product[] {
            return products;
        }
    }

##### Vue

    <!-- tab: App.vue -->
    <template>

>         <DxTreeView
>             // ...

Reviewer 3: ```suggestion
        <DxTreeView ...
```

    }, {
        ID: "1_1_2_5",
        categoryId: "1_1_2",
        name: "SuperLCD 70",
        image: IMAGE_URL + "9.png",
        price: 4000
    }, {
        ID: "1_1_3",
        categoryId: "1_1",
        name: "Monitors"
    }, {
        ID: "1_1_3_1",
        categoryId: "1_1_3",
        name: "19\"",
    }, {
        ID: "1_1_3_1_1",
        categoryId: "1_1_3_1",
        name: "DesktopLCD 19",
        image: IMAGE_URL + "10.png",
        price: 160
    }, {
        ID: "1_1_4",
        categoryId: "1_1",
        name: "Projectors"
    }, {
        ID: "1_1_4_1",
        categoryId: "1_1_4",
        name: "Projector Plus",
        image: IMAGE_URL + "14.png",
        price: 550
    }, {
        ID: "1_1_4_2",
        categoryId: "1_1_4",
        name: "Projector PlusHD",
        image: IMAGE_URL + "15.png",
        price: 750
    }];

    export default products;

##### React

    <!-- tab: App.js -->
    import React from 'react';
    import 'devextreme/dist/css/dx.common.css';
    import 'devextreme/dist/css/dx.light.css';

    import TreeView from 'devextreme-react/tree-view';
    import products from './products';

    function App() {
        return (

>             <TreeView 
>                 ...

Reviewer 3: ```suggestion
            <TreeView ...
```

The TreeView supports plain and hierarchical (default) data structures. In a hierarchical data structure, each node should have a text, unique identifier, and optionally nest other nodes. Refer to the following demo for more information: [Hierarchical Data Structure](/Demos/WidgetsGallery/Demo/TreeView/HierarchicalDataStructure/).

> If a data source has a plain structure, set the [dataStructure](/Documentation/ApiReference/UI_Components/dxTreeView/Configuration/#dataStructure) property to *"plain"*. Each node should reference its child node, specify a text and a unique identifier. Refer to this demo for details: [Plain Data Structure](https://js.devexpress.com/Demos/WidgetsGallery/Demo/TreeView/FlatDataStructure/).

Reviewer 3: The article could end here because everybody went to the demos at this point.

```suggestion
```

The TreeView component can load data from different data source types. To use a local array, assign it to the [items](/Documentation/ApiReference/UI_Components/dxTreeView/Configuration/#items) property. 

To bind the TreeView to a data source, use the [dataSource](/Documentation/ApiReference/UI_Components/dxTreeView/Configuration/#dataSource) property. The following articles explain how to do this for different data structures:

> - [Local Array](/Documentation/Guide/Data_Binding/Specify_a_Data_Source/Local_Array/)

Reviewer 3: ```suggestion
The TreeView component can load data from different data source types. To use a local array, assign it to the [dataSource](/Documentation/ApiReference/UI_Components/dxTreeView/Configuration/#dataSource) or [items](/Documentation/ApiReference/UI_Components/dxTreeView/Configuration/#items) property. If you want to use another data source type, refer to one of the following articles:
```

Author: Considering the issue with two properties, I changed the paragraph to
> "To bind the TreeView to a data source, use the [dataSource]() property. The following articles explain how to do this for different data structures:"

- [Read-Only Data in JSON Format](/Documentation/Guide/Data_Binding/Specify_a_Data_Source/Read-Only_Data_in_JSON_Format/)
- [Web API, PHP, MongoDB](/Documentation/Guide/Data_Binding/Specify_a_Data_Source/Web_API,_PHP,_MongoDB/)
- [OData](/Documentation/Guide/Data_Binding/Specify_a_Data_Source/OData/)
- [Custom Data Sources](/Documentation/Guide/Data_Binding/Specify_a_Data_Source/Custom_Data_Sources/)

> In this tutorial, we use a local hierarchical array. 

Reviewer 3: ```suggestion
The TreeView supports plain and hierarchical data structures. To use plain data, set the [dataStructure](/Documentation/ApiReference/UI_Components/dxTreeView/Configuration/#dataStructure) property to *"plain"*. Each data object should contain fields that specify a node identifier, text to display, and an identifier of the parent node. Use the [keyExpr](/Documentation/ApiReference/UI_Components/dxTreeView/Configuration/#keyExpr), [displayExpr](/Documentation/ApiReference/UI_Components/dxTreeView/Configuration/#displayExpr), and [parentIdExpr](/Documentation/ApiReference/UI_Components/dxTreeView/Configuration/#parentIdExpr) properties to set these data fields:
```

---
##### jQuery

    <!-- tab: index.js -->
    $(function() {
        $("#treeView").dxTreeView({
            dataSource: products,
            dataStructure: "plain",
            keyExpr: "ID",
            displayExpr: "name",
            parentIdExpr: "categoryId",
        });
    });

    <!-- tab: products.js -->
    const IMAGE_URL = "https://js.devexpress.com/Demos/WidgetsGallery/JSDemos/images/products/";

>     const products = [{

Reviewer 3: ```suggestion
    const products = [
        {
```

        name: "SuperLCD 55",
        image: IMAGE_URL + "6.png",
        price: 1750
    }, {
        ID: "1_1_2_5",
        categoryId: "1_1_2",
        name: "SuperLCD 70",
        image: IMAGE_URL + "9.png",
        price: 4000
    }, {
        ID: "1_1_3",
        categoryId: "1_1",
        name: "Monitors"
    }, {
        ID: "1_1_3_1",
        categoryId: "1_1_3",
        name: "19\"",
    }, {
        ID: "1_1_3_1_1",
        categoryId: "1_1_3_1",
        name: "DesktopLCD 19",
        image: IMAGE_URL + "10.png",
        price: 160
    }, {
        ID: "1_1_4",
        categoryId: "1_1",
        name: "Projectors"
    }, {
        ID: "1_1_4_1",
        categoryId: "1_1_4",
        name: "Projector Plus",
        image: IMAGE_URL + "14.png",
        price: 550
    }, {
        ID: "1_1_4_2",
        categoryId: "1_1_4",
        name: "Projector PlusHD",
        image: IMAGE_URL + "15.png",
        price: 750
    }];

    @Injectable({
        providedIn: 'root'
    })
    export class ProductsService {
        getProducts(): Product[] {
            return products;
        }
    }

##### Vue

    <!-- tab: App.vue -->
    <template>
        <DxTreeView
            ...

>             :data-source=products

Reviewer 3: ```suggestion
            :data-source="products"
```

            dataStructure="plain"
            keyExpr="ID"
            displayExpr="name"

>             parentIdExpr="categoryId"

Reviewer 3: Vue syntax uses kebab-case.

        ID: "1_1_2_5",
        categoryId: "1_1_2",
        name: "SuperLCD 70",
        image: IMAGE_URL + "9.png",
        price: 4000
    }, {
        ID: "1_1_3",
        categoryId: "1_1",
        name: "Monitors"
    }, {
        ID: "1_1_3_1",
        categoryId: "1_1_3",
        name: "19\"",
    }, {
        ID: "1_1_3_1_1",
        categoryId: "1_1_3_1",
        name: "DesktopLCD 19",
        image: IMAGE_URL + "10.png",
        price: 160
    }, {
        ID: "1_1_4",
        categoryId: "1_1",
        name: "Projectors"
    }, {
        ID: "1_1_4_1",
        categoryId: "1_1_4",
        name: "Projector Plus",
        image: IMAGE_URL + "14.png",
        price: 550
    }, {
        ID: "1_1_4_2",
        categoryId: "1_1_4",
        name: "Projector PlusHD",
        image: IMAGE_URL + "15.png",
        price: 750
    }];

    export default products;

##### React

    <!-- tab: App.js -->
    import React from 'react';
    import 'devextreme/dist/css/dx.common.css';
    import 'devextreme/dist/css/dx.light.css';

    import TreeView from 'devextreme-react/tree-view';
    import products from './products';

    function App() {
        return (
            <TreeView 

>                 ...

Reviewer 3: ```suggestion
```

Author: There's the id field from the previous step. Thus this ellipsis.

                dataSource={products}
                dataStructure="plain"
                keyExpr="ID"
                displayExpr="name"
                parentIdExpr="categoryId"
            />
        );
    }

    export default App;

    <!-- tab: products.js -->
    const IMAGE_URL = "https://js.devexpress.com/Demos/WidgetsGallery/JSDemos/images/products/";

>     const products = [{

Reviewer 3: ```suggestion
    const products = [
        {
```

            ID: "1_1_2_4",
            categoryId: "1_1_2",
            name: "SuperLCD 55",
            image: IMAGE_URL + "6.png",
            price: 1750
        }, {
            ID: "1_1_2_5",
            categoryId: "1_1_2",
            name: "SuperLCD 70",
            image: IMAGE_URL + "9.png",
            price: 4000
        }, {
            ID: "1_1_3",
            categoryId: "1_1",
            name: "Monitors"
        }, {
            ID: "1_1_3_1",
            categoryId: "1_1_3",
            name: "19\"",
        }, {
            ID: "1_1_3_1_1",
            categoryId: "1_1_3_1",
            name: "DesktopLCD 19",
            image: IMAGE_URL + "10.png",
            price: 160
        }, {
            ID: "1_1_4",
            categoryId: "1_1",
            name: "Projectors"
        }, {
            ID: "1_1_4_1",
            categoryId: "1_1_4",
            name: "Projector Plus",
            image: IMAGE_URL + "14.png",
            price: 550
        }, {
            ID: "1_1_4_2",
            categoryId: "1_1_4",
            name: "Projector PlusHD",
            image: IMAGE_URL + "15.png",
            price: 750
        }
    ];

    export default products;

> ---

Reviewer 3: ```suggestion
---

If your dataset has a hierarchical structure, refer to the following demo for information on how to bind the TreeView to it: [Hierarchical Data Structure](https://js.devexpress.com/Demos/WidgetsGallery/Demo/TreeView/HierarchicalDataStructure/).
```
